mllib/src/main/scala/org/apache/spark/ml/classification/DecisionTreeClassifier.scala

```diff
@@ -144,7 +144,7 @@ final class DecisionTreeClassificationModel private[ml] (
   }
 
   override def toString: String = {
-    s"DecisionTreeClassificationModel of depth $depth with $numNodes nodes"
+    s"${super.toString} DecisionTreeClassificationModel of depth $depth with $numNodes nodes"
```

Comment:
How about using `$uid` directly in `toString` implementation? It is not guaranteed that `super.toString` contains the UID. I suggest `DecisionTreeClassificationModel (uid=$uid) of depth $depth with $numNodes nodes`.